src/dcegm/pre_processing/state_space.py

```diff
@@ -512,3 +530,15 @@ def check_options(options):
         raise ValueError("Model parameters must be a dictionary.")
 
     return options
+
+
+def array_with_smallest_uint_dtype(array):
```

Comment:
Should we check for smallest int type and not uint. Of course if the user gives us only >=0 state vars then it is uint. Then we could additionally support negative state variable values? I think it would not be much work!